nixos/modules/services/system/userborn.nix

```
@@ -166,20 +199,33 @@ in
       };
     };
 
-    # Statically create the symlinks to passwordFilesLocation when they're not
-    # inside /etc because we will not be able to do it at runtime in case of an
-    # immutable /etc!
-    environment.etc = lib.mkIf (cfg.passwordFilesLocation != "/etc") (
-      lib.listToAttrs (
-        lib.map (
-          file:
-          lib.nameValuePair file {
-            source = "${cfg.passwordFilesLocation}/${file}";
-            mode = "direct-symlink";
-          }
-        ) passwordFiles
-      )
-    );
+    environment.etc =
+      if cfg.static then
+        # In static mode, statically drop the files into an immutable /etc.
+        lib.listToAttrs (
+          lib.map (
+            file:
+            lib.nameValuePair file {
+              source = "${userbornStaticFiles}/${file}";
+              mode = if file == "shadow" then "0440" else "0444";
```

This should have the same permissions Userborn would otherwise set:
```suggestion
              mode = if file == "shadow" then "0000" else "0644";
```